pip/req/req_install.py

```diff
@@ -343,11 +344,15 @@ def setup_py(self):
         assert self.source_dir, "No source dir for %s" % self
         try:
             import setuptools  # noqa
-        except ImportError:
+        except ImportError as e:
+            if get_installed_version('setuptools') is None:
+                add_msg = "Please install setuptools."
+            else:
+                add_msg = traceback.format_exc(e)
             # Setuptools is not available
             raise InstallationError(
-                "setuptools must be installed to install from a source "
-                "distribution"
+                "Could not import setuptools which is required to "
+                "install from a source distribution. %s" % add_msg
```

Comment:
I think this should be " ...\n%s" otherwise there's no distinction between the pip message and the add_msg. - and since the exception is formatted as multi line itself we'll want it to be clear.